extension/src/popup/components/sep24/index.tsx

```diff
@@ -0,0 +1,155 @@
+import React, { useState, useEffect } from "react";
+import { useDispatch, useSelector } from "react-redux";
+
+import { startSep24Deposit } from "@shared/api/internal";
+import { getAssetFromCanonical } from "helpers/stellar";
+import { Loader } from "@stellar/design-system";
+
+import { settingsNetworkDetailsSelector } from "popup/ducks/settings";
+import { AppDispatch } from "popup/App";
+import {
+  publicKeySelector,
+  hardwareWalletTypeSelector,
+} from "popup/ducks/accountServices";
+import { LedgerSign } from "popup/components/hardwareConnect/LedgerSign";
+import { useNetworkFees } from "popup/helpers/useNetworkFees";
+import {
+  sep24DataSelector,
+  startHwSign,
+  ShowOverlayStatus,
+  transactionSubmissionSelector,
+  storeSep24Data,
+} from "popup/ducks/transactionSubmission";
+import { openTab } from "popup/helpers/navigate";
+import { Sep24Status } from "popup/constants/sep24";
+import { addTrustline, getAnchorSep24Data } from "popup/helpers/sep24";
+
+import "./styles.scss";
+
+export const Sep24Todo = ({
+  asset,
+  todo,
+  token,
+}: {
+  asset: string;
+  todo: string;
+  token: string;
+}) => {
+  const dispatch: AppDispatch = useDispatch();
+  const { sep10Url, sep24Url, anchorDomain } = useSelector(sep24DataSelector);
+  const networkDetails = useSelector(settingsNetworkDetailsSelector);
+  const publicKey = useSelector(publicKeySelector);
+  const { recommendedFee } = useNetworkFees();
+  const {
+    hardwareWalletData: { status: hwStatus },
+  } = useSelector(transactionSubmissionSelector);
+  const isHardwareWallet = !!useSelector(hardwareWalletTypeSelector);
+  const [isLoading, setIsLoading] = useState(false);
+
+  const { code, issuer } = getAssetFromCanonical(asset);
+
+  useEffect(() => {
+    setIsLoading(false);
+  }, [todo]);
+
+  let message = null;
+  let onClick = () => {};
+
+  switch (todo) {
+    case Sep24Status.INCOMPLETE:
+      message = (
+        <>
+          <div className="Sep24Todo__title">
+            You have an unfinished deposit form
+          </div>
+          <div>click here to finish</div>
```

Comment:
super NIT and you're probably already thinking of this: Might be nice to make this a reusable component

Reply:
yup I think that is better 👍 
https://github.com/stellar/freighter/pull/707/commits/21e047ed8b71569f98b35e4443c65df645ec0669